src/Reflectify/Reflectify.cs

```diff
@@ -609,23 +606,25 @@ public Reflector(Type typeToReflect, MemberKind kind)
         LoadProperties(typeToReflect, kind);
         LoadFields(typeToReflect, kind);
 
-        Members = selectedProperties.Concat<MemberInfo>(selectedFields).ToArray();
+        Members = [.. selectedProperties, .. selectedFields];
```

Comment:
| Method               | N  | Mean       | Error      | StdDev     | Ratio | RatioSD | Gen0   | Code Size | Allocated | Alloc Ratio |
|--------------------- |--- |-----------:|-----------:|-----------:|------:|--------:|-------:|----------:|----------:|------------:|
| CollectionExpression | 1  |   8.733 ns |  0.0387 ns |  0.0302 ns |  0.06 |    0.00 | 0.0064 |     178 B |      40 B |        0.17 |
| Linq                 | 1  | 145.496 ns |  1.4966 ns |  1.3999 ns |  1.00 |    0.01 | 0.0370 |     441 B |     232 B |        1.00 |
|                      |    |            |            |            |       |         |        |           |           |             |
| CollectionExpression | 5  |  25.966 ns |  0.2627 ns |  0.2329 ns |  0.12 |    0.00 | 0.0166 |     187 B |     104 B |        0.35 |
| Linq                 | 5  | 225.200 ns |  2.6079 ns |  2.4394 ns |  1.00 |    0.01 | 0.0470 |     441 B |     296 B |        1.00 |
|                      |    |            |            |            |       |         |        |           |           |             |
| CollectionExpression | 10 |  42.679 ns |  0.4042 ns |  0.3583 ns |  0.13 |    0.00 | 0.0293 |     187 B |     184 B |        0.49 |
| Linq                 | 10 | 331.498 ns |  1.8378 ns |  1.5346 ns |  1.00 |    0.01 | 0.0596 |     441 B |     376 B |        1.00 |
|                      |    |            |            |            |       |         |        |           |           |             |
| CollectionExpression | 20 |  75.985 ns |  0.5488 ns |  0.4583 ns |  0.13 |    0.00 | 0.0548 |     185 B |     344 B |        0.64 |
| Linq                 | 20 | 576.204 ns | 11.3742 ns | 10.6395 ns |  1.00 |    0.03 | 0.0849 |     441 B |     536 B |        1.00 |

<details>
<summary>benchmark code</summary>

```cs
using BenchmarkDotNet.Attributes;
using BenchmarkDotNet.Running;
using System.Collections.Generic;
using System.Linq;
using System.Reflection;
using System.Runtime.InteropServices;

BenchmarkRunner.Run<Benchmark>();

[MemoryDiagnoser]
[DisassemblyDiagnoser]
public class Benchmark
{
    private List<PropertyInfo> selectedProperties;
    private List<FieldInfo> selectedFields;

    [Params(1, 5, 10, 20)]
    public int N { get; set; }

    [GlobalSetup]
    public void Setup()
    {
        selectedProperties = Enumerable.Range(0, N).Select(_ => null as PropertyInfo).ToList();
        selectedFields = Enumerable.Range(0, N).Select(_ => null as FieldInfo).ToList();
    }

    [Benchmark]
    public MemberInfo[] CollectionExpression() =>
        [.. selectedProperties, .. selectedFields];

    [Benchmark(Baseline = true)]
    public MemberInfo[] Linq() =>
        selectedProperties.Concat<MemberInfo>(selectedFields).ToArray();
}
```

</details>

Reply:
Wow. That's a pretty significant improvement. You should post about this on social media (or allow me to do that).

Reply:
Feel free to post about it. 
But be aware that this micro optimization depends on how the compiler lowers the collection expression. A the compiler has freedom in how to lower it as long as it preserves semantics, upgrading the SDK may improve (or regress) a given collection expression 
If you look in the roslyn repository you'll also find issues about collection expressions being lowered to suboptimal C# performance wise.

Reply:
I don't have a lot of interest in those kinds of details, nor the time for it. But fortunately, I have you ;-)

Reply:
Found out that I had already [tweeted](https://twitter.com/jnyrup/status/1739759819873984828) about this one year ago when I did the exact same optimization for `Reflector` when it lived in FA. 

https://github.com/fluentassertions/fluentassertions/commit/0b9566e94776eb2b567a210159c0d674c75efa7c